docs/gravitee.md

```diff
@@ -0,0 +1,96 @@
+# Developer Portal APIs
+
+The Developer Portal is where the Orchestra Cities APIs are registered, and the
+policies defined. The following section describes how these APIs are registered
+in the Developer Portal and how policies and Oauth2 authorization are handled.
+
+## The Authorization flow
+
+1. The user requests a token for the client API from Keycloak. The token comes
+   from a public client in Keycloak, so no secret is required to obtain a token.
+   However it's important to note that the public clients set up in Keycloak will
+   limit what the given tokens enable users to do, such as user management operations.
+   The user requests a specific scope with the request (e.g. entity:read). A list
+   of scopes is available in the **Keycloak Configuration** section of this documentation.
+
+2. When then token is used, it is validated against Keycloak's confidential client
+   (the resource-server client). The operation being attempted (e.g. creating an
+   entity under Tenant X and Servicepath Y)
+   is allowed if the correct scope is present and the Tenant and Servicepath allow
+   it.
+
+## Oauth2 Resource
+
+![Resources](rsrc/keycloak/gravitee_resources.png)
+
+Within the Developer Portal, for each API, Oauth2 access to Keycloak is configured
+as a Generic Oauth2 Authorization Server Resource. This is done to check tokens
+against the confidential client resource-server in Keycloak, where scopes are
+listed.
+
+The configuration contains:
+
+- **Authorization Server URL**: The Keycloak URL
+- **Token introspection endpoint**: /auth/realms/default/protocol/openid-connect/token/introspect
+- **Token introspection method**: POST
+- **Userinfo endpoint**: /auth/realms/default/protocol/openid-connect/userinfo
+- **Userinfo method**: GET
+- **Client Id**: resource-server
+- **Client secret**: The secret of the resource-server client
+- **Authorization header**: Authorization
+- **Authorization scheme**: Basic
+- **Token query param name**: token
+- **Form param name**: token
+
+
+## Policies and Groovy
+
+![Policies](rsrc/keycloak/gravitee_policies.png)
+
+For each API endpoint, policies are in place for each operation. Policies follow
+a basic syntax of resource:operation, so for the /entities endpoint for example
+we have:
+
+- GET requires scope entity:read
```

Comment:
why? i think you need to explain the rational and link to how this is configurable by each api owner based on what he wants to achieve in term of security (and complexity of security controll).